var/spack/repos/builtin/packages/libmesh/package.py

```diff
@@ -19,29 +19,194 @@ class Libmesh(Package):
+                                'sfc', 'tetgen', 'triangle']:
+            if '+' + bundled_library.lower() in self.spec:
+                options.append('--enable-' + bundled_library + "=yes")
+            else:
+                options.append('--enable-' + bundled_library + "=no")
+
+        # and the ones which are dependencies of other bundled libraries:
+        if '+exodusii' in self.spec or '+netcdf' in self.spec:
+            options.append('--enable-netcdf=yes')
+        else:
+            options.append('--enable-netcdf=no')
+
+        # handle external library dependencies:
+        if '+boost' in self.spec:
+            options.append('--with-boost=%s' % self.spec['boost'].prefix)
+        else:
+            options.append('--enable-boost=no')
+
+        if '+eigen' in self.spec:
+            options.append('--with-eigen=%s' % self.spec['eigen'].prefix)
+        else:
+            options.append('--enable-eigen=no')
+
+        if '+hdf5' in self.spec:
+            options.append('--with-hdf5=%s' % self.spec['hdf5'].prefix)
+        else:
+            options.append('--enable-hdf5=no')
+            # This is necessary with libMesh 1.2.1 to prevent a configure
+            # error:
+            if '+netcdf' not in self.spec:
+                options.append('--disable-netcdf-4')
+
+        if '+metis' in self.spec:
+            options.append('--with-metis=PETSc')
+
+        if '+petsc' in self.spec:
+            options.append('--enable-petsc=yes')
+            options.append('PETSC_DIR=%s' % self.spec['petsc'].prefix)
+        else:
+            options.append('--enable-petsc=no')
+
+        if '+slepc' in self.spec:
+            options.append('SLEPC_DIR=%s' % self.spec['slepc'].prefix)
+        else:
+            options.append('--enable-slepc=no')
+
+        # and, finally, other things:
+        if '+debug' in self.spec:
+            options.append('--with-methods=dbg')
+        else:
+            options.append('--with-methods=opt')
+
+        if '+mpi' in self.spec:
+            options.append('CC=%s' % self.spec['mpi'].mpicc)
+            options.append('CXX=%s' % self.spec['mpi'].mpicxx)
+            options.append('--with-mpi=%s' % self.spec['mpi'].prefix)
+        else:
+            options.append('--disable-mpi')
+            # libMesh will try to link with the system MPI library unless we
+            # provide explicit overrides
+            options.append('CC=%s' % self.compiler.cc)
+            options.append('CXX=%s' % self.compiler.cxx)
 
-    def install(self, spec, prefix):
-        config_args = ["--prefix=%s" % prefix]
+        if 'threads=openmp' in self.spec:
+            # OpenMP cannot be used if pthreads is not available: see
+            # parallel/threads_pthread.h and parallel/threads.h
+            options.append('--enable-openmp=yes')
+            options.append('--with-thread-model=pthread')
+            options.append('--enable-pthreads=yes')
+        else:
```

Comment:
With multi False, it would probably be easier to read as

If openmp
  ...
elif pthreads
  .....
elif tbb
else
  ....

Reply:
Unfortunately we need to explicitly add the `--enable-X=no` flag at the end so I believe using `if`/`elif` would just be more complicated.